twine/settings.py

```diff
@@ -0,0 +1,280 @@
+                 "when uploading to PyPI. Other implementations may not "
+                 "support this.)",
+        )
+        parser.add_argument(
+            "--cert",
+            action=utils.EnvironmentDefault,
+            env="TWINE_CERT",
+            default=None,
+            required=False,
+            metavar="path",
+            help="Path to alternate CA bundle (can also be set via %(env)s "
+                 "environment variable).",
+        )
+        parser.add_argument(
+            "--client-cert",
+            metavar="path",
+            help="Path to SSL client certificate, a single file containing the"
+                 " private key and the certificate in PEM format.",
+        )
+        parser.add_argument(
+            "--verbose",
+            action="store_true",
+            help="Show verbose output."
+        )
+
+    @classmethod
+    def from_argparse(cls, args):
+        """Generate the Settings from parsed arguments."""
+        settings = vars(args)
+        settings['repository_name'] = settings.pop('repository')
+        settings['cacert'] = settings.pop('cert')
+        return cls(**settings)
+
+    def _handle_package_signing(self, sign, sign_with, identity):
+        if not sign and identity:
+            raise exceptions.InvalidSigningConfiguration(
+                "sign must be given along with identity"
+            )
+        self.sign = sign
+        self.sign_with = sign_with
+        self.identity = identity
+
+    def _handle_repository_options(self, repository_name, repository_url):
+        self.repository_config = utils.get_repository_from_config(
+            self.config_file,
+            repository_name,
+            repository_url,
+        )
+        self.repository_config['repository'] = utils.normalize_repository_url(
+            self.repository_config['repository'],
+        )
+
+    def _handle_authentication(self, username, password):
+        self.username = utils.get_username(username, self.repository_config)
+        self.password = utils.get_password(
+            self.repository_config['repository'],
+            self.username,
+            password,
+            self.repository_config,
+        )
+
+    def _handle_certificates(self, cacert, client_cert):
+        self.cacert = utils.get_cacert(cacert, self.repository_config)
+        self.client_cert = utils.get_clientcert(
+            client_cert,
+            self.repository_config,
+        )
+
+    def check_repository_url(self):
+        """Verify we are not using legacy PyPI.
+
+        :raises:
+            :class:`~twine.exceptions.UploadToDeprecatedPyPIDetected`
+        """
+        repository_url = self.repository_config['repository']
+
+        if repository_url.startswith((repository.LEGACY_PYPI,
+                                      repository.LEGACY_TEST_PYPI)):
+            raise exceptions.UploadToDeprecatedPyPIDetected(
+                "You're trying to upload to the legacy PyPI site '{0}'. "
```

Comment:
Since this error message is always the same, why not move it into the `UploadToDeprecatedPyPIDetected` class?

Reply:
That's a great idea! I can do that in this PR if you'd like or move it to another.

Reply:
Since you're already touching this code, totally fine to do it here.

Reply:
:white_check_mark: done